src/js/components/FileInput/FileInput.js

```diff
@@ -239,8 +238,32 @@ const FileInput = forwardRef(
         nextFiles.splice(index, 1);
       }
       setFiles(nextFiles);
-      if (onChange) onChange(event, { files: nextFiles });
-      if (nextFiles.length === 0) inputRef.current.value = '';
+
+      // Need to have a way to track the files other than an array
+      // since inputRef.current.files is a read-only FileList
+      // https://stackoverflow.com/a/64019766
+      /* eslint-disable no-undef */
+      const dt = new DataTransfer();
+      const curFiles = inputRef.current.files;
+      if (index === 'all' || nextFiles.length === 0)
+        inputRef.current.value = '';
```

Comment:
Why do we change the `value` in this case but not in the case where there are files?

Reply:
this resets `event.target.files` so we're only doing this in the case where a user wants to remove all of the files or there aren't any files anymore